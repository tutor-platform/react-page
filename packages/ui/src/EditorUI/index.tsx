import React from 'react';
import DisplayModeToggle, { StickyNess } from '../DisplayModeToggle/index';
import ThemeProvider from '../ThemeProvider';
import PluginDrawer from '../PluginDrawer/index';
import Trash from '../Trash/index';

export default React.memo(
  ({
    stickyNess = {
      shouldStickToTop: false,
      shouldStickToBottom: false,
      rightOffset: 0,
    },
    hideEditorSidebar = false,
  }: {
    stickyNess?: StickyNess;
    hideEditorSidebar?: boolean;
  }) => (
    <ThemeProvider>
      <Trash />
      {!hideEditorSidebar && <DisplayModeToggle stickyNess={stickyNess} />}
      <PluginDrawer />
    </ThemeProvider>
  )
);
