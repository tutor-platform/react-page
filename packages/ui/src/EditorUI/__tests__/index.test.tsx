import React from 'react';
import { shallow } from 'enzyme';

import EditorUI from '../';

describe('EditorUI', () => {
  const defaultProps = {
    stickyNotes: {
      shouldStickToTop: false,
      shouldStickToBottom: false,
      rightOffset: 0,
    },
    hideEditorSidebar: false,
  };

  it('renders correctly with default properties', () => {
    const component = shallow(<EditorUI {...defaultProps} />);

    expect(component).toMatchSnapshot();
  });

  it('renders correctly with option to hide the editor sidebar', () => {
    const component = shallow(<EditorUI {...defaultProps} hideEditorSidebar />);

    expect(component.find('DisplayModeToggle')).toHaveLength(0);
  });
});
