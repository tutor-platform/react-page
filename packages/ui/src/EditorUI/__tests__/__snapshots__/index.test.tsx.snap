// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorUI renders correctly with default properties 1`] = `ShallowWrapper {}`;
